lib/ace/edit_session.js

```diff
@@ -324,6 +324,19 @@ var EditSession = function(text, mode) {
     };
 
     /**
+     * @param {Number} row
+     * 
+     * Returns context at row.
+     **/
+    this.getContext = function(row) {
```

Comment:
I suggest make this in the next pull request 
I  would like to draw everything I did as a good class, which will be initiated in the papeeria_mode.js